azkaban-exec-server/src/main/java/azkaban/execapp/ProjectCacheCleaner.java

```diff
@@ -152,9 +159,8 @@ public boolean accept(final File dir, final String name) {
       }
     })) {
```

Comment:
This logic which is pre-existing is flawed, it can cause NPE if there is no match.  This could be handled by,
Objects
        .requireNonNull(this.projectCacheDir.listFiles(new FilenameFilter() {
          String pattern = "[0-9]+\\.[0-9]+";

          @Override
          public boolean accept(final File dir, final String name) {
            return name.matches(this.pattern);
          }
        }))) {

Nit : the indentation needs to be fixed.